### DISCUSSION TITLE
Nous A1T bricked :(

### DISCUSSION THREAD

**Discussion by lennvilardi** [arendst/Tasmota#16244] [Q&A] [answered]
Hi !
I've bought a pack of 4 Nous A1T and I wanted to put all of them up to date (bad habit).
But for one of them something went wrong. 
I tried to reset it holding more than 40s the power button and unplugged it 6 time and let it plugged the 7th.
Still no led signal. So I decided to try to reflash it using a FTDI FT232. I've flashed many Sonoff with it.
The interior design has changed compared of what I've found on the internet (https://templates.blakadder.com/nous_A1.html and https://www.esphome-devices.com/devices/Nous-A1). 
I've found a TX, and I think next to it RX, and the 3.3V. I've tried to put it on programming mode directly on the pin of the ESP8266EX but I'm unsuccessful.
If anyone have an idea... soldering is not very easy on this board for me so i hotglued dupond on the pads. 
Is there a GPIO 0 more accessible on this board ?
![185214216-34d0e6d8-b054-45e3-a3b7-59ea93312d48](https://user-images.githubusercontent.com/10863269/185248721-6727d7bc-9451-4f6a-ac47-5d54dcec5c76.jpg)
![185214220-3e2858e7-2849-452c-bcb8-8d27e116ba27](https://user-images.githubusercontent.com/10863269/185248732-536caa43-ac60-49e9-a15d-8e3ab66cacad.jpg)

**Comment by hagenbuch** [comment]
Has Tasmota been working before? Does any LED go on when plugging it in?

In order to reset Tasmota, you don't need to press any button, just

1. Unplug it for some 30 seconds.
2. Then 6 times: plug it in, wait 5 seconds, pull it out, 5 seconds..
3. Seventh time: Leave it plugged in. The accesspoint mode should start and offer you a "tasmota-something" Wifi

I bought 4 Nous A1 some months ago but their PCB is blue and looks quite different, the chip is 8285! I managed to flash them OTA with this

https://github.com/ct-Open-Source/tuya-convert

**Reply by barbudor** [reply]
Tuya convert is unlikely to work

I flashed 8 or 10 of them plus the similar gosund one
Mine are exactly as the photos on the template web site

Remember that for flashing you need
- gnd
- 3v3 from a good serial adapter, many ft are not suitable
- RX and TX
- gpio0 to gnd before resetting or powering up

By the on those photos there are obviously no RGB led (referring your other question) which would be a square part 3x3mm up to 5x5mm

**Reply by lennvilardi** [reply]
Thanks for the quick answer.
Tasmota have already been working on that plug.
No led at startup.
It's the new version A1T which bring directly a Tasmota firmware.
Are the 5 seconds wait beetween the plug and pull out mandatory ? I will close it again and try it.
I can confirm in the right of RX it's TX (you see the inscription when you lift a bit the relay).
I'll keep you posted

**Reply by lennvilardi** [reply]
> Tuya convert is unlikely to work
> 
> I flashed 8 or 10 of them plus the similar gosund one Mine are exactly as the photos on the template web site
> 
> Remember that for flashing you need
> 
>     * gnd
Found !

> 
>     * 3v3 from a good serial adapter, many ft are not suitable

I've bought a new one but it will come in 2 months
> 
>     * RX and TX
Found !
> 
>     * gpio0 to gnd before resetting or powering up
Maybe it's the part I'm not good at, I use the pinout from https://tasmota.github.io/docs/Pinouts/#esp8266-and-esp8285, but with a dupond it's very hard not touching the others pins. 

> 
> By the on those photos there are obviously no RGB led (referring your other question) which would be a square part 3x3mm up to 5x5mm

There is one led which is green at startup near the power button

**Reply by sfromis** [reply]
When touching gpio0 to ground directly on the ESP chip, do not worry about also touching the neighbor pin to the right. That one is perfectly safe to ground too, and does not affect flashing.

**Comment by barbudor** [comment]
The leds D1 and D4 are single color leds
![image](https://user-images.githubusercontent.com/2996042/185246644-0a9e5142-65c7-4988-bdad-93dd34fff46a.png)
![image](https://user-images.githubusercontent.com/2996042/185246693-f8f6580e-ae73-4f66-bb24-6fae4e4e6804.png)

A RGB led would have at leats 4 pins, whether it is an analog one (3 LEDs in signle package) or a digital one (like AP102, WS2812...).

**Comment by lennvilardi** [comment, marked as answer]
Thanks all for your guidance and I finally nailed it !
My solders are not my best but considering the coating and the tight place ^^
I think the power button is on GPIO 0 because when I press it, it gets directly in programming mode.

![185247442-f64479b4-b92e-4b66-a28c-7b9eec139992](https://user-images.githubusercontent.com/10863269/185248806-7b1d9d94-9316-4ae2-9542-cd51d8313d39.jpg)

**Reply by sfromis** [reply]
The template for Nous A1T indeed indicates that the button is gpio 0, quite different from the previous Nous A1. While externally similar, internally they are unrelated devices.

**Reply by popy2k14** [reply]
thx a lot, this revived my bricked a1t after cross flashing tasmota-minimal.bin over esphome webserver.

**Comment by gabest11** [comment]
I have various smart plugs, now waiting for a few A1T in the mail and found this discussion in google. On the photos, the internals look exactly like my old Neo Coolcam 16a models (NAS-WR01W). I had a few where tuya convert worked and transfered the contents of the flash chip to make the others work, too. No soldering needed, only a hot air gun, a tweezer, and something that can run flashrom. Probably takes me 5 minutes to do one.

**Comment by massa007** [comment]
How did you manage to open that plug? Have exactly the same model, but I cannot figure out how to get the pcb out of the case... the transparent part which is fixed with a screw was the easy part, but now I am struggling because I cannot find any other screw or clip or whatever?

**Reply by barbudor** [reply]
There should be a screw in the earth hole

**Reply by gabest11** [reply]
I used a vice to carefully push the two pins. They are held by the plastic molding. Yea.. it wont be the same again, but the screw will hold it together. Don't forget to carefully remove the wifi antenna, I broke a few.

**Reply by massa007** [reply]
Yep - the vice worked, thanks a lot! I tried pushing by hand in the beginning but it wasn´t moving at all. I guess a hammer would have worked too if used carefully.
In my model was no screw in the earth hole @barbudor, thanks a lot anyway!

**Reply by BlackCube4** [reply]
I actually broke the wifi antenna :( what can I do about it?

**Reply by sfromis** [reply]
Depends on how exactly you broke it. It may work with a wire of the same length.

**Comment by fabianmuehlberger** [comment]
Hi, I have the same problem and currently try to flash the stock espHome on it. 
I am using an FT232 Serial adapter and was able to flash other devices with it. 

Would be very helpful if you could provide instructions on how you managed to make it work. 
Best Fabian

**Reply by Jason2866** [reply]
If the inside is the same as the Neo Coolcam NAS-WR01W it is dangerous and forbidden to use anymore in the EU.  German [link](https://www.bundesnetzagentur.de/SharedDocs/Downloads/DE/Sachgebiete/Telekommunikation/Unternehmen_Institutionen/Technik/InverkehrbringenvonProdukten/EuropaeischeVerbote/PI_220300195252.pdf.pdf?__blob=publicationFile&v=2) this regulation is valid for the whole EU.

**Reply by barbudor** [reply]
Unfortunately the doc doesn't have photos on the inside
I have Nous A1 from before they separated A1 and A1T  which inside (and outside) are the same as my Gosund SP111 (those with the screw in the French earth hole and easy to spot flashing points on the edge of the PCB)

These Néo Coolcam from the outside more looks like the Avatto OT06 which I have in 1 sample but didn't used more than the 10 min required to flash tasmota and calibrate

**Reply by lennvilardi** [reply]
![91604362-ccceca00-e96e-11ea-9e17-f14dd634de6e](https://user-images.githubusercontent.com/10863269/216435792-ce094b42-aa50-4dd1-afa6-3bdd9d7968f8.jpg)
It's not the same plug see on the right side there is R8 and C5 on nas-wr01w and in the nous one there is R1, it's not the same design. I hope they are safe to us

**Reply by barbudor** [reply]
My Avatto is more or less exactly the same
I recognize the spots where I soldered the wires for the flashing
I can't tell for the rest if it's exactly the same

Your coil seems to be damaged

**Reply by lennvilardi** [reply]
It's not mine I found it on another github thread just to compare if they were identical. 
They're not but I don't found any document of conformity for the nous a1t plug on their website, they only displayed a CE label which may not be true...

**Reply by sfromis** [reply]
CE is no certification, really not more than the manufacturer saying *belive me*, this product conforms. Hard to go after manufacturers in China in case of "differing" ideas about what conformance is.

**Comment by Fettkeewl** [comment]
> Hi !
> I've bought a pack of 4 Nous A1T

Hey!
Sorry for necroing this. may I ask how you opened the A1T? Screw under sticker or force the edges (glued?)
I have some coil whine id like to try and get rid of

**Reply by barbudor** [reply]
If this model doesn't have the screw in the French earth hole, you need to remove the sticker on the female side and unscrew to remove the cover
Then press firmly to disangage the male pins that will push all the block out
But don't go too fast too far or you may damage the antenna cable

**Reply by meganom** [reply]
Yes, there is a triangular screw under sticker:
![20230223_215111](https://user-images.githubusercontent.com/2887318/221038136-2f4fec86-d5b2-440d-9058-43d6e63352f8.jpg)

And yes, there is no screw in the French earth hole. I noticed it only in A1 version.

**Comment by meganom** [comment]
Just for information. I have also a problem with this plug, described here: https://github.com/ascillato/Tasmota_KNX/discussions/595
This post helped a lot to understand where to connect serial interface. 

@lennvilardi  Have you faced any new problems after re-flashing via serial interface? And which serial adapter have you used, FTDI FT232? Thank you!

**Comment by Jason2866** [comment]
@meganom  The issue you have linked are not related.

**Comment by Miq1** [comment]
Could someone give me a hint where to find the TX pin on the A1T? RX, 3V3 and GND are obvious, and GPIO00 is connected to the button switch - but where is TX?

**Reply by meganom** [reply]
Hi!

It is here:
![image](https://user-images.githubusercontent.com/2887318/222771986-d5abc2ca-dc18-4d62-9240-1099a9600d7f.png)

Tx label is under Relais. So be careful not to damage the plastic case of the relais.

**Reply by Miq1** [reply]
The D1 and D4 pads are open - I suppose the former two red and blue LEDs were supposed to go there? Anyone tried to solder some SMD LEDs on those pads?

**Comment by thomas-lentz** [comment]
Hi,
I know I shouldn't do this, but it is the most technical thread I found to Nous A1T...
The current Nous A1T configuration (according to blakadder's template) doesn't "see" the direction of the current. If you use the Nous A1T in conjunction with a solar-inverter there is no "-" sign at the Power or the current, when you supply to the house/grid. To see if there is any software solution to this (without opening the Nous A1T), I would like to know, which power measurement chipset is used in the Nous A1T. unfortunately pictures here don't allow me to read it.
I appreciate your help!

Greetings Tom

**Reply by Silverstar** [reply]
The chip, according to the pictures on the esphome page @lennvilardi posted, is a BL0942.
I don't know if it is possible to receive a signed value, as I did not research on that chip.

**Reply by Silverstar** [reply]
Seems to be that it is possible but was reverted due to some bug, see #15265 and #15374.

**Reply by barbudor** [reply]
If I remember well, the reason was that some devices were starting to errorneously show negative values because of bad hardware design by manufacturer (cabling was inverted on some devices), not because of a Tasmota bug. 
So it was safer to revert the changes

**Reply by thomas-lentz** [reply]
Yes, I found this [thread #15374](https://github.com/arendst/Tasmota/issues/15374) too, but wouldn't it be nice (and wouldn't it be the right way) to have a current-direction option: "always positive", "always negative", "invert the direction: meaning toggle the '-/+'" with default "always positive".
This would allow the "ease of use for dummies", compensating bad designed devices and the two direction current/power metering.

**Reply by Silverstar** [reply]
What about a template change like BL0942 and BL0942_i ?

**Reply by thomas-lentz** [reply]
Well, when you use the Nous A1T to measure the output (and input) of a Solar Inverter, there is a little problem with just inverting.
In the nighttime the solar inverter will draw a little current from the grid and if you do abs() you can't distinguish anymore between consuming and providing energy.

**Reply by barbudor** [reply]
Just use the right tool for the right usage

**Reply by thomas-lentz** [reply]
Was this really the best suggestion you could come up with or are you just bored on this sunny sunday afternoon :-)?!?
I guess I already [pointed out](https://github.com/arendst/Tasmota/discussions/16244#discussioncomment-5335965) what the correct implementation of the software (Tasmota) should be for adapters like the Nous A1T, Gosund SP111, ... and so on. 
Just because some manufacturers are too stupid to construct a correct device and just because some users are not able to RTFM, you normally don't create a solution/software with reduced functionality, you work with adequate defaults.
The Nouse A1 or alike adapters are perfectly suited to do this job for a small solar installation, why don't you think so?

**Reply by barbudor** [reply]
A plug is not meant and designed to be used in that situation.
You're just tweaking the use-case.
When it comes to pure power measurments, there are better solutions.

Now, you're free to propose an implementation that would work with either a SetOption or a `BL0942_i` GPIO.

**Reply by thomas-lentz** [reply]
Okay, when I'm seeing this right: both settings (`BL0942` & `BL0942_i`) do an abs() for the meassured value, right?
What about an additional `BL0942_d `and `BL0942_di` ("d" for direction-sensitive), which just omits the abs() after meassuring?

Unfortunately the solution with "_i" contradicts a _wholistic solution_ and more genereal solution (in regards to different Power-Meassurement chips) with a SetOption, which could look like this
```
0: "always positive" (default, today's behaviour)
1: "always negative" (like the "_i" solution)
2: "direction-sensitive" (show current flow/powerconsumption as designed by the power meassurement chip producer) 
3: "direction-sensitive inverted" (the same as "2" but inverted)
```

**Reply by arendst** [reply]
Choose a SetOption as this uses less resources.

**Reply by arendst** [reply]
BTW the ``-i`` option is not meant for positive/negative measurements. It's meant for hardware designers using an inverted signal.

**Reply by barbudor** [reply]
I do agree. My suggestion was actually
- implement a SetOption to say 'energy inverted' 
- remove the abs() and use the value of the setoption flag to actually change sign if set

I'm not sure we can avoid a breaking change as some hardware implementation may be correct and some not. So some will need to activate the setoption.

**Reply by thomas-lentz** [reply]
Why not make it the suggested way
```
Setoption<yy> [0..3]
0: "always positive" (default and also today's behaviour)
1: "always negative" (like today's behaviour with a "-" in front of it  / value multiplied by -1)
2: "direction-sensitive" (show current flow/powerconsumption as designed by the power meassurement chip producer) 
3: "direction-sensitive inverted" (the same as "2" but inverted / value multiplied by -1)

```
Having the "0" as the default, the behaviour would be like today - avoiding any breaking change. The "1" would be the opposite sign of today's display. "2" and "3" you choose according to your specific device ...

**Reply by Jason2866** [reply]
The + / - direction is not needed. The power drawn from the Inverter is under 1W. This device cannot measure such low loads. So every discussion needing +/- direction for connected Microinverter is an academic one.

**Comment by r10r** [comment]
Has someone of you added some additional sensors to the Nous A1T already ?
I'd like to attach one or more [DS18b20](https://esphome.io/components/sensor/dallas.html) temperature sensors and/or the [BME280 Environment sensor](https://esphome.io/cookbook/bme280_environment.html?highlight=bme280)

Is it possible reuse RX / TX or the D1/D4 pad  as IO pins for I2C  ?
I'll update the https://devices.esphome.io/devices/Nous-A1t page if possible.

**Reply by gabest11** [reply]
I did this with a power strip (Blitzwolf BW-SHP9), but it did not work realiably, if the I2C device was plugged in, it failed to boot (correction: soft reboot, hard boot was somehow okay, iirc), you have to examine the required boot states. Another thing I learned, a temperature sensor is completely useless inside a closed space, the micro generates heat and it gets trapped in.

**Reply by barbudor** [reply]
The Nous A1T includes power measurement circuit which is NOT isolated from mains (ie the ESP GPIOs may be at AC level) 
As such it is highly discouraged to make any modification especially those that involve getting some wires out of the box to connect external sensors.

**Reply by r10r** [reply]
Thanks for your response. What pins did you use for I2C - the RX/TX pins or D1 or D4 ?

>  Another thing I learned, a temperature sensor is completely useless inside a closed space, the micro generates heat and it gets trapped in.

~~I plan to drill a little hole and attach the sensor  with a wire to measure the temperature inside of the freezer.~~

**Reply by r10r** [reply]
@barbudor Thanks for the response - so I'll dismiss this idea. 
Maybe we should add this warning to the device's page https://devices.esphome.io/devices/Nous-A1t,
because the pinout inspires people (like me) to attach sensors to it.

**Reply by gabest11** [reply]
That's a very good question, I can't remember, it was maybe a year ago. It is still wired in but not very eager to disassemble it right now :)

**Reply by r10r** [reply]
@barbudor Can you recommend a device where it is possible to attach external sensors ?

**Reply by barbudor** [reply]
It all depends which form factor you are looking for.

If it's a plug, then only ones without power measurments would most likely be safe but infortunately most are not as easy to open as the Nous A1T.

Shelly Plus 1 have connectors for external sensors: https://kb.shelly.cloud/knowledge-base/shelly-plus-1#ShellyPlus1(SNSW-001X16EU)-Addoninterface

Otherwise, for sensors build, the simplest is a NodeMCU/Mini-D1 and a USB wall adapter

**Comment by rira2005** [comment]
HI
Thanks your Solution work for me. Button GPIO0!
Used Tasmotizer to unbricking. ;-)
THX

**Comment by StefanFanaru** [comment]
Hi
I'm a total beginner to electronics tinkering in general.
Do you have any idea if this outlet can be flashed using a CH341A?
If so, any tips on how?
I bricked mine while trying to upgrade the firmware, tried everything to get it back working, the light on the button does not turn on no matter what. I have a CH341A on hand so it got me thinking.

**Reply by sfromis** [reply]
CH341A cannot be recommended for a "total beginner", but you can find stuff via Google. Here's one using it to read the flash chip along an ESP8266, same tools can be used to write the chip. 
https://www.youtube.com/watch?v=rkmjRJZX7fU
OTOH, if you *really* want to get into using the CH341A, handling a flash chip used with ESP8266 is not a bad starting point for the learning curve. And Google is your friend.....

A more beginner-friendly approach: https://tasmota.github.io/docs/Getting-Started/
Specifically for Nous A1T, this thread already has good info.

**Reply by StefanFanaru** [reply]
> CH341A cannot be recommended for a "total beginner", but you can find stuff via Google. Here's one using it to read the flash chip along an ESP8266, same tools can be used to write the chip. https://www.youtube.com/watch?v=rkmjRJZX7fU OTOH, if you _really_ want to get into using the CH341A, handling a flash chip used with ESP8266 is not a bad starting point for the learning curve. And Google is your friend.....
> 
> A more beginner-friendly approach: https://tasmota.github.io/docs/Getting-Started/ Specifically for Nous A1T, this thread already has good info.

Thank you for your reply! I appreciate it!
I’m a bit lost and don’t know where to start.
I have the CH341A and a Raspberry PI. 
If i want to flash the firmare on this outlet what should i do? Do i have to solder wires on it? That’s what i’m seeing in this post. Is the raspberry pi more usefull for this task? Or can i use the CH341A clam

**Reply by sfromis** [reply]
My recommendation would be the let the CH341A rest to another time, unless your primary purpose is to learn to use it. In that case, "Google is your friend", as there is "some learning curve" with tools and so on.

What was already discussed in this thread is indeed to solder a few wires, and use a standard USB/serial adapter, as recommended in the last link I posted, having the procedure for flashing.

As you "only" need serial, it should be *possible* to use the Pi too, without such an adapter, as long as you are "up to speed" in setting up serial, and also making sure to supply sufficient 3.3V power (at least 150 mA) during flashing.
